api/v1alpha1/httproute_types.go

```diff
@@ -192,7 +192,52 @@ type HTTPRouteAction struct {
 
 // HTTPRouteStatus defines the observed state of HTTPRoute
 type HTTPRouteStatus struct {
-	Gateways []core.ObjectReference `json:"gateways"`
+	// Gateways is a list of the Gateways that are associated with the
+	// HTTPRoute, and the status of the route with respect to each of these
+	// Gateways.  Each associated Gateway should have exactly one entry in
+	// this list.
+	Gateways []RouteGatewayStatus `json:"gateways"`
+}
+
+// RouteGatewayStatus describes the status of a route with respect to an
+// associated Gateway.
+type RouteGatewayStatus struct {
+	// Gateway is a reference to a Gateway object that is associated with
+	// the route.
+	Gateway core.ObjectReference `json:"gateway"`
```

Comment:
This PR has `core.ObjectReference` for consistency with master; if #73 is accepted and this PR is not rejected, I will update this field's type.